src/libraries/System.Net.WebSockets/tests/WebSocketDeflateTests.cs

```diff
@@ -0,0 +1,343 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.WebSockets.Tests
+{
+    public class WebSocketDeflateTests
+    {
+        private readonly CancellationTokenSource? _cancellation;
+
+        public WebSocketDeflateTests()
+        {
+            if (!Debugger.IsAttached)
+            {
+                _cancellation = new CancellationTokenSource(TimeSpan.FromSeconds(5));
+            }
+        }
+
+        public CancellationToken CancellationToken => _cancellation?.Token ?? default;
+
+        public static IEnumerable<object[]> SupportedWindowBits
+        {
+            get
+            {
+                for (var i = 9; i <= 15; ++i)
+                {
+                    yield return new object[] { i };
+                }
+            }
+        }
+
+        [Fact]
+        public async Task HelloWithContextTakeover()
+        {
+            WebSocketTestStream stream = new();
+            stream.Enqueue(0xc1, 0x07, 0xf2, 0x48, 0xcd, 0xc9, 0xc9, 0x07, 0x00);
```

Comment:
All the tests with specific byte representation do the check for reading (ReceiveAsync). Maybe it will also make sense to add the same tests but for writing - that write into stream, and then it is checked for resulting bytes?